dex/fiatrates/sources.go

```diff
@@ -0,0 +1,246 @@
+package fiatrates
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"decred.org/dcrdex/dex"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const (
+	defaultRefreshInterval = 5 * time.Minute
+	messariRefreshInterval = 10 * time.Minute
+
+	cryptoCompare              = "CryptoCompare"
+	cryptoComparePriceEndpoint = "https://min-api.cryptocompare.com/data/pricemulti?fsyms=%s&tsyms=USD"
+
+	binance                = "Binance"
+	binancePriceEndpoint   = "https://api3.binance.com/api/v3/ticker/price?symbols=[%s]"
+	binanceUSPriceEndpoint = "https://api.binance.us/api/v3/ticker/price?symbols=[%s]"
+
+	coinpaprika              = "Coinparika"
+	coinpaprikaPriceEndpoint = "https://api.coinpaprika.com/v1/tickers"
+
+	messari              = "Messari"
+	messariPriceEndpoint = "https://data.messari.io/api/v1/assets/%s/metrics/market-data"
+
+	kuCoin              = "KuCoin"
+	kuCoinPriceEndpoint = "https://api.kucoin.com/api/v1/prices?currencies=%s"
```

Comment:
Comments here summarizing what you know about the api request limits and potentially what type of errors we would see if we go over. How did you end up with those values for refresh interval?

Reply:
> Comments here summarizing what you know about the api request limits 

Okay. 

>  How did you end up with those values for refresh interval?

I'll add in the comments.

Reply:
Missed this in my latest push: https://github.com/decred/dcrdex/pull/2736/commits/8f9328ff68c76e3f5de337db2d8218be285078eb

I will have to circle back to this later. Just remembered I posted most of it in the matrix channel sometime ago but we should have as much information here too for debug purpose in the future.

Reply:
Updated in https://github.com/decred/dcrdex/pull/2736/commits/dbe0d1bf43a49d89df60fa84af09b668f505355f